math/minuit2/StandAlone.cmake

```diff
@@ -52,21 +52,28 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
     set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
     set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
     set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+    # Avoid name collisions when used as a subproject
+    set(Minuit2MathTarget Math)
+    set(Minuit2CommonTarget Common)
+else()
+    set(Minuit2MathTarget Minuit2Math)
+    set(Minuit2CommonTarget Minuit2Common)
```

Comment:
Any particular reason not to always use `Minuit2{Math,Common}` names?

Reply:
Because then the exported targets become `Minuit2::Minuit2Math`. Honestly though, now that I'm looking at them, I'm not sure anyone is supposed to be linking to those at all; in which case, export names don't really matter.

Reply:
I see. `Minuit2::Math` does read better, but we cannot export them with different name than the library? For instance, openssl has `OpenSSL::SSL` and `OpenSSL::Crypto` exported, but I think those come from `FindOpenSSL.cmake` not from the build system itself.

Reply:
You are not supposed to write a `Find*` package for a CMake project that provides exported targets. Those are two different things. I don't think you can export with different names, but I'm not completely sure. I'll check.

Reply:
I know, that's why I wondered if it was possible to call `export` or `install(EXPORT ...)` and change the target name that gets exported...

Reply:
That should be better. I'm not sure if we really need to provide the simple name targets, but if that's the case, I can just remove the 6 lines in the Config file.

Reply:
Changing the name is also nice because the file created was `libMath.a`, which is a terrible name for a library. (That can be changed without changing the name of the target, but this now makes it a much more reasonable `libMinuit2Math.a` by default)

Reply:
> Changing the name is also nice because the file created was libMath.a

Exactly. That was my original concern, `libMinuit2Math.*` is better if the exported target can be renamed. Also, you could add an alias target `Math` to `Minuit2Math` and export that, right?

Reply:
I've done that. It can't actually be an ALIAS target, due to techincal restrictions about exporting and aliases on imported targets, but I've created an IMPORTED INTERFACE target in the `Config.cmake.in` that links to the original target (for Math and Common). So from a user's standpoint, nothing has changed, `Minuit2::Math` is still available regardless of the usage method (`add_subdirectory` or `find_package`)

Reply:
Very nice!